### ISSUE TITLE
I can't get training to start.

### ISSUE THREAD

**Issue by Deejay85** [bmaltais/kohya_ss#396]
I'm trying to train a new fetish using Lora, and while I've been watching some videos on how to set the basic training parameters, despite doing everything I'm supposed to, it's just not working. Also, while I did watch another video in case that made things easier, I didn't gain any new information on what I'm doing wrong. While I am new to all of this I do know that the files are supposed to go into the "100_name here" folder, as well as having a model and log folder all in the same directory. 

If it helps any I've tried training with the latest version of Waifu Diffusion, and when that didn't work, I tried Lewd Diffusion and R34...both gave the same results. Not sure if it helps any, but there are multiple checkpoints in the stable-diffusion folder. and yes, I made sure to choose the file manually other than just setting the root directory.

```
Load CSS...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Folder 100_Huge Balls: 2200 steps
max_train_steps = 1100
stop_text_encoder_training = 0
lr_warmup_steps = 0
accelerate launch --num_cpu_threads_per_process=2 "train_network.py" --v2 --v_parameterization --pretrained_model_name_or_path="S:/WaifuDiffusion/models/Stable-diffusion/Waifu Diffusion.ckpt" --train_data_dir="S:/kohya_ss/Sample Images/Lora Folder/Image" --resolution=512,512 --output_dir="S:/kohya_ss/Sample Images/Lora Folder/Model" --logging_dir="S:/kohya_ss/Sample Images/Lora Folder/Log" --network_alpha="128" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-5 --unet_lr=0.0001 --network_dim=128 --output_name="huge balls" --lr_scheduler_num_cycles="1" --learning_rate="0.0001" --lr_scheduler="constant" --train_batch_size="2" --max_train_steps="1100" --save_every_n_epochs="1" --mixed_precision="bf16" --save_precision="bf16" --seed="1234" --caption_extension=".txt" --cache_latents --optimizer_type="AdamW8bit" --max_data_loader_n_workers="1" --clip_skip=2 --bucket_reso_steps=64 --xformers --bucket_no_upscale
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_processes` was set to a value of `1`
        `--num_machines` was set to a value of `1`
        `--mixed_precision` was set to a value of `'no'`
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
v2 with clip_skip will be unexpected / v2でclip_skipを使用することは想定されていません
prepare tokenizer
Use DreamBooth method.
prepare images.
found directory S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls contains 22 image files
2200 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 512)
  enable_bucket: False

  [Subset 0 of Dataset 0]
    image_dir: "S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls"
    image_count: 22
    num_repeats: 100
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    is_reg: False
    class_tokens: Huge Balls
    caption_extension: .txt


[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 22/22 [00:00<00:00, 4400.53it/s]
prepare dataset
prepare accelerator
Traceback (most recent call last):
  File "S:\kohya_ss\train_network.py", line 652, in <module>
    train(args)
  File "S:\kohya_ss\train_network.py", line 108, in train
    accelerator, unwrap_model = train_util.prepare_accelerator(args)
  File "S:\kohya_ss\library\train_util.py", line 1973, in prepare_accelerator
    accelerator = Accelerator(gradient_accumulation_steps=args.gradient_accumulation_steps, mixed_precision=args.mixed_precision,
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 370, in __init__
    raise ValueError(err.format(mode="bf16", requirement="PyTorch >= 1.10 and a supported device."))
ValueError: bf16 mixed precision requires PyTorch >= 1.10 and a supported device.
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "S:\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 45, in main
    args.func(args)
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 1104, in launch_command
    simple_launcher(args)
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 567, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['S:\\kohya_ss\\venv\\Scripts\\python.exe', 'train_network.py', '--v2', '--v_parameterization', '--pretrained_model_name_or_path=S:/WaifuDiffusion/models/Stable-diffusion/Waifu Diffusion.ckpt', '--train_data_dir=S:/kohya_ss/Sample Images/Lora Folder/Image', '--resolution=512,512', '--output_dir=S:/kohya_ss/Sample Images/Lora Folder/Model', '--logging_dir=S:/kohya_ss/Sample Images/Lora Folder/Log', '--network_alpha=128', '--save_model_as=safetensors', '--network_module=networks.lora', '--text_encoder_lr=5e-5', '--unet_lr=0.0001', '--network_dim=128', '--output_name=huge balls', '--lr_scheduler_num_cycles=1', '--learning_rate=0.0001', '--lr_scheduler=constant', '--train_batch_size=2', '--max_train_steps=1100', '--save_every_n_epochs=1', '--mixed_precision=bf16', '--save_precision=bf16', '--seed=1234', '--caption_extension=.txt', '--cache_latents', '--optimizer_type=AdamW8bit', '--max_data_loader_n_workers=1', '--clip_skip=2', '--bucket_reso_steps=64', '--xformers', '--bucket_no_upscale']' returned non-zero exit status 1.
```

**Comment by aztack** [comment]
Same issue here. I'm on master branch (v21.2.4).
Then I switch back to v21.23, there is nothing happed after click 'Train Model' button.
Then switch to v21.21, still nothing happed. No feedback, not error in the console.

**Comment by bmaltais** [comment]
did you complete the `accelerate config` step? Try running that again.

**Comment by Deejay85** [comment]
Having some trouble here. The file exists at...copied and pasted it into Powershell with the following command:

```
PS S:\kohya_ss> S:\kohya_ss\venv\Scripts\accelerate-config.exe update
Sucessfully updated the configuration file at C:\Users\<REDACTED>/.cache\huggingface\accelerate\default_config.yaml.
```


The file's location however is at s:\kohyaa_ss, so why is it appearing there, I've no idea. Also, when I click on training, I get this message:


```
Validating that requirements are satisfied.
All requirements satisfied.
Load CSS...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Loading config...
Folder 100_Huge Balls: 2200 steps
max_train_steps = 1100
stop_text_encoder_training = 0
lr_warmup_steps = 0
accelerate launch --num_cpu_threads_per_process=2 "train_db.py" --v2 --pretrained_model_name_or_path="S:/WaifuDiffusion/models/Stable-diffusion/Waifu Diffusion.ckpt" --train_data_dir="S:/kohya_ss/Sample Images/Lora Folder/Image" --resolution=512,512 --output_dir="S:/kohya_ss/Sample Images/Lora Folder/Model" --logging_dir="S:/kohya_ss/Sample Images/Lora Folder/Log" --save_model_as=safetensors --output_name="huge balls" --max_data_loader_n_workers="1" --learning_rate="0.0001" --lr_scheduler="constant" --train_batch_size="2" --max_train_steps="1100" --save_every_n_epochs="1" --mixed_precision="bf16" --save_precision="bf16" --seed="1234" --caption_extension=".txt" --cache_latents --optimizer_type="AdamW8bit" --max_data_loader_n_workers="1" --clip_skip=2 --bucket_reso_steps=64 --xformers --bucket_no_upscale
v2 with clip_skip will be unexpected / v2でclip_skipを使用することは想定されていません
prepare tokenizer
prepare images.
found directory S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls contains 22 image files
2200 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 512)
  enable_bucket: False

  [Subset 0 of Dataset 0]
    image_dir: "S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls"
    image_count: 22
    num_repeats: 100
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    is_reg: False
    class_tokens: Huge Balls
    caption_extension: .txt


[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 22/22 [00:00<00:00, 3666.93it/s]
prepare dataset
prepare accelerator
Traceback (most recent call last):
  File "S:\kohya_ss\train_db.py", line 364, in <module>
    train(args)
  File "S:\kohya_ss\train_db.py", line 75, in train
    accelerator, unwrap_model = train_util.prepare_accelerator(args)
  File "S:\kohya_ss\library\train_util.py", line 1984, in prepare_accelerator
    accelerator = Accelerator(gradient_accumulation_steps=args.gradient_accumulation_steps, mixed_precision=args.mixed_precision,
  File "S:\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 370, in __init__
    raise ValueError(err.format(mode="bf16", requirement="PyTorch >= 1.10 and a supported device."))
ValueError: bf16 mixed precision requires PyTorch >= 1.10 and a supported device.
Traceback (most recent call last):
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\<REDACTED>\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "S:\kohya_ss\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "S:\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 45, in main
    args.func(args)
  File "S:\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 1104, in launch_command
    simple_launcher(args)
  File "S:\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 567, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['S:\\kohya_ss\\kohya_ss\\venv\\Scripts\\python.exe', 'train_db.py', '--v2', '--pretrained_model_name_or_path=S:/WaifuDiffusion/models/Stable-diffusion/Waifu Diffusion.ckpt', '--train_data_dir=S:/kohya_ss/Sample Images/Lora Folder/Image', '--resolution=512,512', '--output_dir=S:/kohya_ss/Sample Images/Lora Folder/Model', '--logging_dir=S:/kohya_ss/Sample Images/Lora Folder/Log', '--save_model_as=safetensors', '--output_name=huge balls', '--max_data_loader_n_workers=1', '--learning_rate=0.0001', '--lr_scheduler=constant', '--train_batch_size=2', '--max_train_steps=1100', '--save_every_n_epochs=1', '--mixed_precision=bf16', '--save_precision=bf16', '--seed=1234', '--caption_extension=.txt', '--cache_latents', '--optimizer_type=AdamW8bit', '--max_data_loader_n_workers=1', '--clip_skip=2', '--bucket_reso_steps=64', '--xformers', '--bucket_no_upscale']' returned non-zero exit status 1.
```

**Comment by bmaltais** [comment]
Change the bf16 to fp16 in the training parameters.

**Comment by Deejay85** [comment]
I don't mind learning new things, but when **IT'S HARDER THAN CHINESE ARITHMETIC** it just gets tiring after a bit. I reinstalled the program to get the latest version, jumped through all the necessary hoops to get it back in working order, deleted and re-created the file default_config by running accelerate-config, and it still didn't work. The solution to my problem? Throwing my computer out of an open window. Just kidding of course. 😜

I ran the program, went to settings, Dreambooth Lora, training parameters, and made the change there...program finally started running. All this time...just that one simple value made it not working. My only question is why? Anyway, for now, I'm going to let it run overnight to see what happens.

**Comment by bmaltais** [comment]
@Deejay85 Don't despair. It look like your card dows not support bitsandbytes for AdamW8bit. Change the optimizer to AdamW instead. That should get you passed this. If NVidia supported the same feature on all it's cards it would make things way simpler.

**Comment by Deejay85** [comment]
How do you know what my graphics card can support? Also discovered something interesting...I updated my drivers...but didn't reboot my computer. I couldn't even pull the Nvidia control panel up. >_< Did a restart...and it didn't fix anything. Well...I tried at least. @_@

Anyway, did try AdamW, and when that didn't work, I tried the rest of them...none of them worked. To make it easier I will just include the instance where I ran AdamW. Also...github **really** needs a spoiler tag to compress long sections of code. >_<


[spoiler]
```
To create a public link, set `share=True` in `launch()`.
Loading config...
Folder 100_Huge Balls: 22 images found
Folder 100_Huge Balls: 2200 steps
max_train_steps = 1100
stop_text_encoder_training = 0
lr_warmup_steps = 0
accelerate launch --num_cpu_threads_per_process=2 "train_network.py" --v2 --pretrained_model_name_or_path="S:/WaifuDiffusion/models/Stable-diffusion/Waifu Diffusion.ckpt" --train_data_dir="S:/kohya_ss/Sample Images/Lora Folder/Image" --resolution=512,512 --output_dir="S:/kohya_ss/Sample Images/Lora Folder/Model" --logging_dir="S:/kohya_ss/Sample Images/Lora Folder/Log" --network_alpha="128" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-5 --unet_lr=0.0001 --network_dim=128 --output_name="huge balls" --lr_scheduler_num_cycles="1" --learning_rate="0.0001" --lr_scheduler="constant" --train_batch_size="2" --max_train_steps="1100" --save_every_n_epochs="1" --mixed_precision="fp16" --save_precision="fp16" --seed="1234" --caption_extension=".txt" --cache_latents --optimizer_type="AdamW" --max_data_loader_n_workers="1" --clip_skip=2 --bucket_reso_steps=64 --xformers --bucket_no_upscale
NOTE: Redirects are currently not supported in Windows or MacOs.
[W ..\torch\csrc\distributed\c10d\socket.cpp:558] [c10d] The client socket has failed to connect to [www.007guard.com]:29500 (system error: 10049 - The requested address is not valid in its context.).
[W ..\torch\csrc\distributed\c10d\socket.cpp:558] [c10d] The client socket has failed to connect to [www.007guard.com]:29500 (system error: 10049 - The requested address is not valid in its context.).
v2 with clip_skip will be unexpected / v2でclip_skipを使用することは想定されていません
prepare tokenizer
Use DreamBooth method.
prepare images.
found directory S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls contains 44 image files
4400 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 512)
  enable_bucket: False

  [Subset 0 of Dataset 0]
    image_dir: "S:\kohya_ss\Sample Images\Lora Folder\Image\100_Huge Balls"
    image_count: 44
    num_repeats: 100
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    is_reg: False
    class_tokens: Huge Balls
    caption_extension: .txt


[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 22/22 [00:00<00:00, 4400.53it/s]
prepare dataset
prepare accelerator
[W ..\torch\csrc\distributed\c10d\socket.cpp:558] [c10d] The client socket has failed to connect to [www.007guard.com]:29500 (system error: 10049 - The requested address is not valid in its context.).
[W ..\torch\csrc\distributed\c10d\socket.cpp:558] [c10d] The client socket has failed to connect to [www.007guard.com]:29500 (system error: 10049 - The requested address is not valid in its context.).
Traceback (most recent call last):
  File "S:\kohya_ss\train_network.py", line 699, in <module>
    train(args)
  File "S:\kohya_ss\train_network.py", line 119, in train
    accelerator, unwrap_model = train_util.prepare_accelerator(args)
  File "S:\kohya_ss\library\train_util.py", line 2498, in prepare_accelerator
    accelerator = Accelerator(
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 308, in __init__
    self.state = AcceleratorState(
  File "S:\kohya_ss\venv\lib\site-packages\accelerate\state.py", line 150, in __init__
    torch.distributed.init_process_group(backend="nccl", **kwargs)
  File "S:\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 602, in init_process_group
    default_pg = _new_process_group_helper(
  File "S:\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 727, in _new_process_group_helper
    raise RuntimeError("Distributed package doesn't have NCCL " "built in")
RuntimeError: Distributed package doesn't have NCCL built in
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 37816) of binary: S:\kohya_ss\venv\Scripts\python.exe
```
[/spoiler]

**Comment by Deejay85** [comment]
Yes...several times...ditto for deleting the default_config.yaml file. Willing to try again if you got any advice.

**Comment by bmaltais** [comment]
It is complaining about socket access. Might be some sort of 3rd party antivirus blocking the access? Do you run a non Microsoft AV on your PC?

**Comment by Deejay85** [comment]
I tried a few things...using a modded .dll file, (`https://rentry.org/2chAI_LoRA_Dreambooth_guide_english`) disabling Avast for 10 minutes, and even downloading Stable Diffusion 2-1 from the quick pick menu as a test case (10 GB download...there's 1 1/2 hours I'll never get back again), all of which were total duds.

I'm running out of ideas here, and copying and pasting `CUDA_LAUNCH_BLOCKING=1` into the extra arguments section of the training parameters isn't helping either...though it is generating an error message that says "got an unexpected keyword argument." 😩

**Comment by bmaltais** [comment]
There is something weird on your system. The socket error is for a domain that does not even exist: www.007guard.com

It is hard to tell what is the root cause... but I am inclined to say that it is external to kohya@ss and related to other software installed on your computer.

**Comment by Deejay85** [comment]
Okay...after a **LOT** of trial and error that surprisingly didn't involve learning Chinese arithmetic, I finally managed to figure out what was wrong by brute-forcing the situation (trying everything I could think of to see what would stick). Turns out all I had to do was disable xformers. What? Clearly, something isn't right here.

I did install the recent Cuda package for Windows 10, got an error, and had to uninstall both drivers for the 1060 GTX and the Tesla M40, reinstall them, and then installed the Cuda package...I got no idea what I'm doing wrong. Also, somehow, I was able to get Lora to run on Automatic1111 for a bit. I'm going to look into trying to troubleshoot xformers right now, because I really don't know what else to do at this point.
